### PR TITLE
Add the plan of what test flakes bot should work.

### PR DESCRIPTION
**Pull request by incfly**

### CODE REVIEW

File: policybot/TEST_FLAKES_BOT.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - `time.Now - last_updates_timestamp` > threath_hold, 3 days.
+- Update the issue with comments "@user-id testing flakes needs your attetion."
+- Send email to the assignee? Maybe later?
+
+## Code Changes
+
+- `plugins/flakeschaser` as pkg.
+- Read spanner table to figure out the working set.
+- Process working item by sending comments as Github client, Github throttler
+might be needed.
```
Comment (geeknoid) [inline policybot/TEST_FLAKES_BOT.md]: If you follow the same pattern as in the rest of the code, you end up throttled naturally. The part that doesn't work is that I haven't implemented the 5000 QPH limit, and the limit is for a given access token so if there are multiple instances hitting GH at the same time, we might hit quota limits.

I don't have a strategy in place to deal with getting quota errors. It's probably worth having some standard way to do this systematically.

File: policybot/TEST_FLAKES_BOT.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- Might better to update the Spanner tables right away to reduce the chance
+another server instance duplicates `flakes chaser` comments. Alternatively, just invoke `/sync` by itself to simplfy the code logic?
+
+## Questions
+
+- What's size of the data to be processed?
+- How are Cloud scheduler invoking the function? Cloud PubSub as glue or what?
```
Comment (geeknoid) [inline policybot/TEST_FLAKES_BOT.md]: I haven't actually set up Cloud Scheduler yet. I was going to try that this weekend.

File: policybot/TEST_FLAKES_BOT.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+Need IAM permission to see the istio-testing Cloud Scheduler job...
```
Comment (geeknoid) [inline policybot/TEST_FLAKES_BOT.md]: I'll make sure you can access the project.

File: policybot/TEST_FLAKES_BOT.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+- Local development, add a `make serve` option to run locally? Haven't find any
+Cloud specific dependency yet...
```
Comment (geeknoid) [inline policybot/TEST_FLAKES_BOT.md]: I have a trivial run.sh script which passes the secrets as command-line arguments. The bot works fine locally and can be run under a debugger.

**Comment by geeknoid** [comment]
You're already a member of the istio-testing@prod.google.com MDB group, so you should be able to access the istio-testing GCP project.

I've just set up Cloud Scheduler to ping the /sync endpoint. It's just sending an HTTP GET request to policybot.istio.io/sync and every night at 2:00 AM. That's all it needs...

File: policybot/TEST_FLAKES_BOT.md (hunk 3 above)
Comment (incfly) [reply policybot/TEST_FLAKES_BOT.md]: I'm already a istio-testing google internel group, but when I look at the project, it appears that i don't have the edit/owner permission to modify the resources. e.g. i can't create cloud scheduler before you set it up.

Comment (incfly) [reply policybot/TEST_FLAKES_BOT.md]: same thing for the deploying cloud run instance.

File: policybot/TEST_FLAKES_BOT.md (hunk 1 above)
Comment (incfly) [reply policybot/TEST_FLAKES_BOT.md]: sg. How about my other thought, 

> - Might better to update the Spanner tables right away to reduce the chance
another server instance duplicates `flakes chaser` comments. Alternatively, just invoke `/sync` by itself to simplfy the code logic?

without this, spanner `issuecomments` table can lag behind right after we post updates to neg people. If before that happened, we might neg people twice.

we can also not worry about this initially until it becomes a problem...

Comment (geeknoid) [reply policybot/TEST_FLAKES_BOT.md]: I've got a PR pending that makes many improvements in this area.

'syncer' tries to sync everything as before. It should run once a day.

'refresher' listens to GitHub events as they occur and then updates Spanner. This means that the syncer should usually be close to a NOP.

In that world, we just need to make the refresher plugin run before any other GitHub plugins so ensure that Spanner is up to date enough. I think anyway.

**Comment by incfly** [comment]
This is no longer needed, i figured, superseded by https://github.com/istio/bots/pull/17

**Pull request closed without merging**